cpp/velox/shuffle/VeloxShuffleWriter.cc

```diff
@@ -623,16 +622,22 @@ arrow::Status VeloxShuffleWriter::buildPartition2Row(uint32_t rowNum) {
   rowOffset2RowId_.resize(rowNum);
   for (auto row = 0; row < rowNum; ++row) {
     auto pid = row2Partition_[row];
-    PREFETCHT0((rowOffset2RowId_.data() + partition2RowOffset_[pid] + 32));
     rowOffset2RowId_[partition2RowOffset_[pid]++] = row;
   }
 
-  std::transform(
-      partition2RowOffset_.begin(),
-      std::prev(partition2RowOffset_.end()),
-      partition2RowCount_.begin(),
-      partition2RowOffset_.begin(),
-      [](uint32_t x, uint32_t y) { return x - y; });
+  for (auto pid = 0; pid < numPartitions_ - 1; ++pid) {
+    partition2RowOffset_[pid] -= partition2RowCount_[pid];
+  }
+
+  // calc valid partition list
+  partitionUsed.clear();
+  for (auto pid = 0; pid != numPartitions_; ++pid) {
+    if (partition2RowCount_[pid] > 0) {
+      partitionUsed.push_back(pid);
+    }
+  }
+
+  std::cout << "[zcw] partitionUsed.size:" << partitionUsed.size() << " numPartitions_:" << numPartitions_ << std::endl;
```

Comment:
Please remove this log, or change it to debug log if necessary.

Reply:
OK